.github/workflows/snapshot-release.yml

```diff
@@ -50,4 +52,121 @@ jobs:
             Version: `${{ steps.set-snapshot-version.outputs.VERSION }}`
             Repository: `https://s01.oss.sonatype.org/content/repositories/snapshots/`
           edit-mode: replace
-          token: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+  # This job depends on the publish job to finish so we can use the snapshot version to verify
+  # gradle compatibility of the sample apps with actual releases of the SDK with latest changes.
+  gradle-compatibility-builds:
```

Comment:
Keeping this with snapshot build ensures we test apps with changes actually published. If we used local, we first need to publish locally with actual AGP version, switch AGP version and then build sample apps. Using snapshot build reduces this effort by not publishing SDK every time separately.